core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -67,50 +67,42 @@ public static void check(final Settings settings) {
      * @param enforceLimits true if the checks should be enforced or
      *                      warned
      * @param checks        the checks to execute
+     * @param nodeName      the node name to be used as a logging prefix
      */
     // visible for testing
-    static void check(final boolean enforceLimits, final List<Check> checks) {
-        final ESLogger logger = Loggers.getLogger(BootstrapCheck.class);
-
-        for (final Check check : checks) {
-            final boolean fail = check.check();
-            if (fail) {
-                if (enforceLimits) {
-                    throw new RuntimeException(check.errorMessage());
-                } else {
-                    logger.warn(check.errorMessage());
-                }
+    static void check(final boolean enforceLimits, final List<Check> checks, final String nodeName) {
+        final ESLogger logger = Loggers.getLogger(BootstrapCheck.class, nodeName);
+
+        final List<IllegalStateException> exceptions =
+                checks.stream()
+                        .filter(BootstrapCheck.Check::check)
+                        .map(check -> new IllegalStateException(check.errorMessage()))
+                        .collect(Collectors.toList());
+
+        if (!exceptions.isEmpty()) {
+            final List<String> messages = new ArrayList<>(1 + exceptions.size());
+            messages.add("bootstrap checks failed");
+            exceptions.forEach(e -> messages.add(e.getMessage()));
+            if (enforceLimits) {
+                final RuntimeException re = new RuntimeException(String.join("\n", messages));
+                exceptions.forEach(re::addSuppressed);
+                throw re;
+            } else {
+                messages.forEach(message -> logger.warn(message));
```

Comment:
I knew what you meant and I don't see it as cleaner, but you forgot to add the `errors` to `messages`. Either way, I pushed a63dcb0ef306a7ff4dbfa8ffd41e6400bccdee77.